cmd/helm/root.go

```diff
@@ -292,10 +297,26 @@ func newDefaultRegistryClient(plainHTTP bool) (*registry.Client, error) {
 	return registryClient, nil
 }
 
-func newRegistryClientWithTLS(certFile, keyFile, caFile string, insecureSkipTLSverify bool) (*registry.Client, error) {
+func newRegistryClientWithTLS(
```

Comment:
I'm looking at the `NewRegistryClientWithTLS` function. It's only ever called from here. But since it's public we can't remove it until Helm 4. I wonder though, could we consolidate some of the logic in that function, in `newRegistryClientWithTLS`, and the logic wrapping the few places where these are called?

Reply:
I think this can be done. We’d leave `NewRegistryClientWithTLS` as just a wrapper calling `newRegistryClientWithTLS` like you suggested

Reply:
@scottrigby now I remember why I could not DRY this code previously. I'd need to expose a new public function (i.e `NewRegistryClientWithTLSWithCreds`) in `pkg/registry/util` for it to be called in `cmd/helm` since we cannot add new parameters to `NewRegistryClientWithTLS`. Copying code over to `cmd/helm` seemed like the lesser evil

Reply:
That seems reasonable to me 👍